packages/app/src/outbox/services/outbox.service.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import {
+  Contact,
+  Endpoint,
+  GetEntriesResponseBody,
+  Method,
+  OutboxCategory,
+  OutboxEntry,
+  OutboxEntryChange,
+  OutboxEntryType,
+} from "@mudita/pure"
+import { MainProcessIpc } from "electron-better-ipc"
+import { asyncNoop } from "Renderer/utils/noop"
+import { DeviceService } from "App/backend/device-service"
+import { IpcEvent } from "App/data-sync/constants"
+import DeviceResponse, {
+  DeviceResponseStatus,
+} from "Backend/adapters/device-response.interface"
+import { ContactRepository } from "App/data-sync/repositories"
+
+export class OutboxService {
+  // @ts-ignore
+  private entryHandlersMap: Record<
+    OutboxEntryType,
+    (entry: OutboxEntry) => Promise<void>
+  > = {
+    [OutboxEntryType.Contact]: this.handleContactEntry.bind(this),
+    // TODO: add Thread, Message handlers
+  }
+
+  constructor(
+    private ipc: MainProcessIpc,
+    private deviceService: DeviceService,
+    private contactRepository: ContactRepository
+  ) {}
+
+  public async readOutboxEntries(): Promise<void> {
+    const { status, data } = await this.getOutboxEntriesRequest()
+
+    const entries = data?.entries
+
+    if (status !== DeviceResponseStatus.Ok || entries === undefined) {
+      return
+    }
+
+    if (entries.length === 0) {
+      return
+    }
+
+    for (const entry of entries) {
+      const handle = this.entryHandlersMap[entry.type] ?? asyncNoop
+      await handle(entry)
+    }
+
+    await this.deleteOutboxEntriesRequest(entries.map(({ uid }) => uid))
+
+    await this.ipc.sendToRenderers(IpcEvent.DataLoaded)
```

Comment:
I think this event can be emitted from the observer

Reply:
It's reduce amount of dependencies in this service

Reply:
[8eb4cd0](https://github.com/mudita/mudita-center/pull/670/commits/8eb4cd036c1d4b0110ede1de9e75605d73a4f95f)